apps/designer/app/designer/shared/floating-panel/floating-panel.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverHeader,
+  PopoverPortal,
+} from "@webstudio-is/design-system";
+import { MutableRefObject, useContext, useRef, useState } from "react";
+import { FloatingPanelContext } from "./floating-panel-provider";
+
+const useSideOffset = ({
+  isOpen,
+}: {
+  isOpen: boolean;
+}): [MutableRefObject<HTMLButtonElement | null>, number] => {
+  const triggerRef = useRef<HTMLButtonElement | null>(null);
+  const { container } = useContext(FloatingPanelContext);
+  if (isOpen === false) {
+    // Prevent computation if the panel is closed
+    return [triggerRef, 0];
+  }
+  const containerRect = container?.current?.getBoundingClientRect();
+  const triggerRect = triggerRef.current?.getBoundingClientRect();
+  return [triggerRef, (triggerRect?.left ?? 0) - (containerRect?.left ?? 0)];
```

Comment:
Accessing dom in render is not a good idea. Refs can still not be set. Use effects

Reply:
done